javarosa/src/main/java/org/javarosa/core/model/data/IntegerData.java

```diff
@@ -50,49 +34,42 @@ public IntegerData(Integer n) {
         setValue(n);
     }
 
+    @Override
     public IAnswerData clone() {
         return new IntegerData(n);
     }
 
-    /* (non-Javadoc)
-     * @see org.javarosa.core.model.data.IAnswerData#getDisplayText()
-     */
+    @Override
     public String getDisplayText() {
         return String.valueOf(n);
     }
 
-    /* (non-Javadoc)
-     * @see org.javarosa.core.model.data.IAnswerData#getValue()
-     */
+    @Override
     public Object getValue() {
         return DataUtil.integer(n);
     }
 
+    @Override
     public void setValue(Object o) {
-        if (o == null) {
-            throw new NullPointerException("Attempt to set an IAnswerData class to null.");
-        }
-        n = ((Integer)o).intValue();
+        n = (Integer)o;
```

Comment:
I actually reverted this later in the PR because I think it was an incorrect change